packages/sdk/src/modules/base/index.ts

```diff
@@ -194,14 +209,44 @@ export class Base {
                     const extinsics: GenericExtrinsic<AnyTuple>[] = (
                       await api.rpc.chain.getBlock(blockHeader.hash)
                     ).block.extrinsics;
-                    const apiAt = await api.at(blockHeader.hash);
-                    const events: any = await apiAt.query?.['system']?.[
-                      'events'
-                    ];
+                   
+                   // these events don't have extrinsic details
+                   // const apiAt = await api.at(blockHeader.hash);
+                    // const events: any = await apiAt.query?.['system']?.[
+                    //   'events'
+                    // ];
+ 
 
+                    const events = await result.events; // get events
```

Comment:
Above line did not properly get the event data for extrinsics. Now using result.events to get the events to create a mapping and filter to check for an "ExtrinsicFailed" event. If one is found, _transactionError gets called to build the failure message, and an Error is thrown.